src/core/egg-herbie.rkt

```diff
@@ -67,13 +67,16 @@
   (free (FFIRule-right rule))
   (free rule))
 
-;; the first hash table maps all symbols and non-integer values
-;; to new names for egg; the second hash is the reverse of the first
-(struct egraph-data (egraph-pointer egg->herbie-dict herbie->egg-dict))
+;; Wrapper around Rust-allocated egg runner
+(struct egraph-data
+        (egraph-pointer ; FFI pointer to runner
+         herbie->egg-dict ; map from symbols to canonicalized names
+         egg->herbie-dict ; inverse map
+         id->spec)) ; map from e-class id to an approx-spec or #f
 
 ; Makes a new egraph that is managed by Racket's GC
 (define (make-egraph)
-  (egraph-data (egraph_create) (make-hash) (make-hash)))
+  (egraph-data (egraph_create) (make-hash) (make-hash) (make-hash)))
```

Comment:
What’s this for?

Reply:
Hacky way to extract approx nodes; essentially save the original approx specs. The alternative would be to add a spec extractor to typed extraction.